src/main/java/uk/gov/hmcts/reform/slc/services/steps/getpdf/FileNameHelper.java

```diff
@@ -11,12 +13,12 @@ public static String generateName(Letter letter, String extension) {
         return letter.getType() + SEPARATOR + letter.getService() + SEPARATOR + letter.getId() + "." + extension;
     }
 
-    public static String extractId(String fileName) {
+    public static UUID extractId(String fileName) {
         String[] parts = FilenameUtils.removeExtension(fileName).split(SEPARATOR);
         if (parts.length != 3) {
             throw new UnableToExtractIdFromFileNameException();
         } else {
-            return parts[2];
+            return UUID.fromString(parts[2]);
```

Comment:
what if it's `"foo"`? This should probably throw `UnableToExtractIdFromFileNameException` exception as well

Reply:
also, a test would be nice.